### ISSUE TITLE
Assembly-CSharp.dll breaks game. Update Folder breaks game

### ISSUE THREAD

**Issue by RetroGameClub** [higurashi-mod/watanagashi#66]
Windows 8 Machine
Followed every step in the Install Instructions TO THE LETTER on a fresh install of the game. Created voice folder, moved voices into it, moved all appropriate cg and cgalt folders, installed streaming assets update patch, installed everything according to the instructions in your guide with no liberties taken and it breaks the game to the level of not even turning on anymore, when running the .exe it won't even start the game anymore.

 I narrowed it down to a file in the "managed" folder called Assembly-CSharp.dll which if you overwrite it with your patched version the game won't even start, but even if you don't transfer that file what happens is the game will load a black screen and never do anything. Then pressing alt+f4 will show the "are you sure" dialogue box and exit.

So basically if you install any of the files in the update folder or the "Managed" folder it completely breaks the game. I successfully moved all cg and voices into the folder and it does not break the game, but when installing the scripts in the update folder this is when the problem happens. Yet no one online has talked about this and the only way I even got the Onikakushi chapter to work properly is by using someone else's streaming assets folder.

Is there any way you can please upload your complete patched streaming assets folder to google drive or megaupload? Because seriously I have literally tried 50 different methods with no results other than a black screen.

**Comment by enumag** [comment]
Never had this problem. I even tested Watanagashi myself right before the release.

>  I successfully moved all cg and voices into the folder and it does not break the game, but when installing the scripts in the update folder this is when the problem happens. 

Our new scripts require the updated DLL so obvoiously the game will fail go to black screen quite soon because of unknown commands.

Do you have the latest version of the game?

Am I correct to assume the game didn't even load the menu?

Did you try our installer? At this point it is more reliable since manual approach requires too many steps.

When you try it again (with the DLL), can you upload the `output_log.txt` file? Should be located in HigurashiEp02_Data.

**Comment by enumag** [comment]
I just tried again on my PC and everything works. Running the game for the first time takes a few seconds longer but that's normal because it compiles the scripts. It could take longer for you depending on the performance of your PC but no longer than half a minute.

**Comment by RetroGameClub** [comment]
Thanks very much for your prompt response. Maybe it's a windows 8 issue...
I tried the auto installer, everything transfers successfully to the proper
folders but game will not run it won't even open the window like it's
attempting to run. Then if I delete that .dll and replace it with the
original one from the game it will run again but only a black screen.

Same thing when trying manual install. The patch for chapter 1 for some
reason did not break the game but the voices didn't work until I downloaded
someone else's streaming assets folder as I mentioned earlier, then
everything worked.

Chapter 2 on the other hand won't even start anymore. It may be because I
have a different version of the game? I didn't realize it had been updated,
I'll look into that as well

I'll get the output from the log file tomorrow and upload it


On Oct 27, 2017 10:54 PM, "Jáchym Toušek" <notifications@github.com> wrote:

Never had this problem. I even tested Watanagashi myself right before the
release.

I successfully moved all cg and voices into the folder and it does not
break the game, but when installing the scripts in the update folder this
is when the problem happens.

Our new scripts require the updated DLL so obvoiously the game will fail go
to black screen quite soon because of unknown commands.

Do you have the latest version of the game?

Am I correct to assume the game didn't even load the menu?

Did you try our installer?

When you try it again (with the DLL), can you upload the output_log.txt
file? Should be located in HigurashiEp02_Data.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/07th-mod/watanagashi/issues/66#issuecomment-340141622>,
or mute the thread

**Comment by enumag** [comment]
@RetroGameClub I do not know if the game was updated recently or not. It's just that some very old versions are known to be incompatible. I don't think this is the reason in your case though.

Sadly I really don't know what could have caused this. Hopefully the log will reveal something.

**Comment by P-Chang** [comment]
I assume game stop in compiling scripts for some reason. because his system run with other user's streaming assets folder.

@enumag 
Do you know the guys that use Win8 system in team or user? my system is Win7, I can't test it in Win8 system.

@RetroGameClub 
At first, please run game with patched AssemblyCSharp.dll and upload output.log file.

I want to know.

1.Your "Higurashi 02 - Watanagashi" folder have folder that is named like "2017-09-18_230530" ? If AssemblyCSharp.dll is broken, game automatically make this.

2.Have you ever use previous version,?

3.Did you installed game in D drive or later and disabled UAC? C drive and UAC have many trouble in modded game.

4.Which store did you buy it? Steam or MG?

5.Please tell me file version and product version of HigurashiEp02.exe in property.

6.In the end, game with patced AssemblyCSharp.dll can run after deleting entire file in "Update" and "CompiledUpdateScripts"? please don't delete folder, game can't launch without these folder.

**Comment by enumag** [comment]
@P-Chang I have no idea what system the rest of our team is using. I have Win 10 so I can't try on Win 8 either. Besides if it works on both Win 7 and Win 10 I really doubt it would be a Win 8 specific issue.

**Comment by P-Chang** [comment]
@ItaloKnox 
@DoctorDiablo 
@Grelo 
@EliEron
@MyLegGuy 

If you use Win8, please tell me status in current version.

**Comment by ItaloKnox** [comment]
I'm currently running Windows 10 Anniversary, so I don't think I can help. My tests using this build were all successful, though.

**Comment by EliEron** [comment]
I do have access to a Windows 8.1 machine and I do a lot of technical troubleshooting as part of my day job so I decided to investigate the problem a bit.

I have access to both the Steam and MG version of the game so I decided to test both versions, what I found was that the patch worked fine on Windows 8.1 when using the Steam version of the game.

The MG version of the game is another story, that crashed instantly just like RetroGameClub reported, however after a bit of testing I found that it crashes not only on Windows 8 but on other platforms as well.

I have linked to the full output_log file at the bottom of the post but I'll paste in one bit that seems pretty telling of the underlying issue:

`[Steamworks.NET] Could not load [lib]steam_api.dll/so/dylib. It's likely not in the correct location. Refer to the README for more details.`

It seems that the patched DLLs are looking for steam_api.dll which makes sense if they were taken from the Steam version of the game, this file is of course not present on the MG version of the game though. And it seems like its presence is required for the patched game to launch.

This issue is not specific to Wataganashi either. I tested the MG version of Onikakushi as well and that had the exact same issue, so I suspect this problem affects all MG versions of Higurashi.

[output_log.txt](https://github.com/07th-mod/watanagashi/files/1424316/output_log.txt)

**Comment by ItaloKnox** [comment]
I guess we narrowed it down, then. I think I have a couple of chapters I bought from MangaGamer directly. Should I upload the .dll? @P-Chang 

Maybe the vanilla steam_api would work? I have access to the Steam SDK and all the developer stuff and as far as I know, if it requires only the presence of the file it might not do any harm. I can give it some testing in a bit if necessary.

**Comment by P-Chang** [comment]
@ItaloKnox 
OK, I'll look over MG DLL, Thanks.

I tested that delete CSteamwroks.dll in chapter1, I got same log and crashed.
And deleting steam_api.dll didn't cause anything in launch.

@EliEron 
Thank you for looking over it. It is really helpful.

**Comment by ItaloKnox** [comment]
Ok, I did some testing and it seems our new dll calls Steam everytime, even if you have a legit steam_api.dll file from the Steam game inside the folder. Running the drm-free game with the steam game installed will actually run the steam game or uninstalling the steam game and running the drm-free game will prompt you to download the steam game instead.

I only have the dll from Chapter 4 so that's the best I can offer for now.
[Assembly-CSharp.zip](https://github.com/07th-mod/watanagashi/files/1424361/Assembly-CSharp.zip)

**Comment by enumag** [comment]
@P-Chang It would be best if you could create a single DLL that would work in both cases. Of course I'm not sure if that's even possible.

Do we have some kind of temporary workaround?

If needed I have access to MG versions of Oni, Wata, Tata and Hima so I can get you their DLLs. I don't have MG version of Mea though.

@RetroGameClub TL;DR version for you: We found the problem and it's that our current patch is only compatible with Steam verions of the games and not MG versions. @P-Chang is working on a fix.

**Comment by ItaloKnox** [comment]
Just wondering if we can't have our current dll just skip the steam_api.dll file in case it doesn't find it? That would be better than having multiple versions of the same file.

**Comment by P-Chang** [comment]
Chpater2 DLL for MG Version test is here
http://www.mediafire.com/file/2i4alggutcf3vh1/Chapter2DLL_FOR_MGVer.rar

I just deleted 2 methods that is not included in MG version DLL.

probably, both dll are same without these 2 method, but Decompiler didn't show the same result(bit different local variable name, sorting, etc), so DIFF with DIFF tool is impossible, and there are possibility that it is not enough.

On my windows7 steam version with deleted every steam dll is working fine.

If @enumag or @RetroGameClub  or @EliEron can handle it in chapter2 MG ver , I'll patch same sequence in every chapter. it is not hard task.

And I bit against releasing deleted 2 method version for steam user, bacause it definitely is deleted method for steam. and I didn't understand it is for what.

And I want to know, 
```
Fallback handler could not load library D:/Program Files (x86)/Steam/steamapps/common/Higurashi When They Cry/HigurashiEp01_Data/Mono/CSteamworks
Fallback handler could not load library D:/Program Files (x86)/Steam/steamapps/common/Higurashi When They Cry/HigurashiEp01_Data/Mono/.\CSteamworks
Fallback handler could not load library D:/Program Files (x86)/Steam/steamapps/common/Higurashi When They Cry/HigurashiEp01_Data/Mono/CSteamworks
Fallback handler could not load library D:/Program Files (x86)/Steam/steamapps/common/Higurashi When They Cry/HigurashiEp01_Data/Mono/libCSteamworks
Fallback handler could not load library D:/Program Files (x86)/Steam/steamapps/common/Higurashi When They Cry/HigurashiEp01_Data/Mono/.\libCSteamworks
Fallback handler could not load library D:/Program Files (x86)/Steam/steamapps/common/Higurashi When They Cry/HigurashiEp01_Data/Mono/libCSteamworks
 
```

Are these log written in MG version with vanilla DLL? After I deleted 2 method, log say this.

**Comment by EliEron** [comment]
@P-Chang I can confirm that those lines do appear in the log when you run the vanilla MG version.

The MG version of Watanagashi does run with the DLL you just uploaded.

I agree that it would likely be a bad idea to use this DLL for Steam users though. While I have not tested this I find it highly likely that any Steam specific feature like Trophies and Cards will not unlock with the modified DLL.

**Comment by enumag** [comment]
@P-Chang Would it be possible to detect availability of the Steam DLL inside those two methods so that we can only have one version of the Assembly-CSharp.dll?

**Comment by P-Chang** [comment]
@enumag 
This DLL work with both version. I want not to release this for steam user mean I can't have responsibility for user.

I found another way that deleting commands that will exit game forcibly if there are no steam dlls. probably it didn't spoil Trophies and Card. But do you really want to use DLL that have edited steamworks method on steam game MODs? I won't.

@EliEron 
Thank you for report. You made this problem to totally clear.

**Comment by ItaloKnox** [comment]
Changing the way the dll talks to Steam won't probably have any issues. All the Steam communication is handled by steam_api, including achievements, cloud saves, etc. It should be safe to deploy a universal dll as long as we can confirm it doesn't break any of the Steam features or directly modify the steam_api file.

If you can give me a dll for Chapter 4 using the method where it ignores the absence of the steam_api file I can give it a proper testing. See if the features are properly working and everything. I'm a little concerned about having multiple versions of the same dll for every game. By the end of the project, we'll probably have 20 or more dlls to update.

**Comment by P-Chang** [comment]
Here is chapter4 DLLs with new way.
http://www.mediafire.com/file/46d8bq5i588z639/Chapter4_MG_Test.rar

Now game can avoid steam with deleting all steamworks dlls.

**Comment by ItaloKnox** [comment]
Thanks, I'll test with both versions and let you know if there's something wrong or if they're ready for production.

**Comment by RetroGameClub** [comment]
ok sorry for the delay guys here is the output of the file you requested on a fresh install of higurashi Chapter 2 after applying the patch with the instructions in the wiki to the letter -
```

Initialize engine version: 5.2.2f1 (3757309da7e7)
GfxDevice: creating device client; threaded=1
Direct3D:
    Version:  Direct3D 11.0 [level 11.0]
    Renderer: NVIDIA GeForce GT 755M  (ID=0xfcd)
    Vendor:   NVIDIA
    VRAM:     1987 MB
Begin MonoManager ReloadAssembly
Platform assembly: C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\UnityEngine.dll (this message is harmless)
Loading C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\UnityEngine.dll into Unity Child Domain
Platform assembly: C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\Assembly-CSharp-firstpass.dll (this message is harmless)
Loading C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\Assembly-CSharp-firstpass.dll into Unity Child Domain
Platform assembly: C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\Assembly-CSharp.dll (this message is harmless)
Loading C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\Assembly-CSharp.dll into Unity Child Domain
Platform assembly: C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\UnityEngine.UI.dll (this message is harmless)
Loading C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\UnityEngine.UI.dll into Unity Child Domain
Platform assembly: C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\UnityEngine.Networking.dll (this message is harmless)
Loading C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\UnityEngine.Networking.dll into Unity Child Domain
Platform assembly: C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\NVorbis.dll (this message is harmless)
Loading C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\NVorbis.dll into Unity Child Domain
Platform assembly: C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\Antlr3.Runtime.dll (this message is harmless)
Loading C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\Antlr3.Runtime.dll into Unity Child Domain
Platform assembly: C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\System.Core.dll (this message is harmless)
- Completed reload, in  0.049 seconds
<RI> Initializing input.

<RI> Input initialized.

desktop: 1920x1080 59Hz; virtual: 1920x1080 at 0,0
<RI> Initialized touch support.

Platform assembly: C:\Gaems\Higurashi2\Higurashi 2 Watanagashi\HigurashiEp02_Data\Managed\System.dll (this message is harmless)
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
[Steamworks.NET] Could not load [lib]steam_api.dll/so/dylib. It's likely not in the correct location. Refer to the README for more details.
System.DllNotFoundException: C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll

  at (wrapper managed-to-native) Steamworks.NativeMethods:SteamAPI_RestartAppIfNecessary (Steamworks.AppId_t)

  at Steamworks.SteamAPI.RestartAppIfNecessary (AppId_t unOwnAppID) [0x00000] in <filename unknown>:0 

  at Assets.Scripts.Core.SteamWorks.SteamManager.Awake () [0x00000] in <filename unknown>:0 
 
(Filename: C:/buildslave/unity/build/artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 64)

Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
DllNotFoundException: C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
  at (wrapper managed-to-native) Steamworks.NativeMethods:SteamAPI_RegisterCallback (intptr,int)

  at Steamworks.Callback`1[Steamworks.UserStatsReceived_t].Register (Steamworks.DispatchDelegate func) [0x00000] in <filename unknown>:0 

  at Steamworks.Callback`1[Steamworks.UserStatsReceived_t]..ctor (Steamworks.DispatchDelegate func, Boolean bGameServer) [0x00000] in <filename unknown>:0 

  at Assets.Scripts.Core.SteamWorks.SteamController.Awake () [0x00000] in <filename unknown>:0 
 
(Filename:  Line: -1)

NullReferenceException: Object reference not set to an instance of an object
  at Assets.Scripts.Core.GameSystem.OnApplicationQuit () [0x00000] in <filename unknown>:0 
 
(Filename:  Line: -1)

Setting up 2 worker threads for Enlighten.
  Thread -> id: 12ac -> priority: 1 
  Thread -> id: 1150 -> priority: 1 
NullReferenceException: Object reference not set to an instance of an object
  at Assets.Scripts.Core.GameSystem.OnDestroy () [0x00000] in <filename unknown>:0 
 
(Filename:  Line: -1)

Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/.\libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
Fallback handler could not load library C:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Mono/libC:/Gaems/Higurashi2/Higurashi 2 Watanagashi/HigurashiEp02_Data/Plugins/CSteamworks.dll
```

**Comment by RetroGameClub** [comment]
Using these instructions on the wiki to install - 

> 
How to install the patch

    Find and open your game folder (open your Steam library, choose the game from the list, right click on it and click on Properties. Switch to the LOCAL FILES tab and click on Browse local files...)
    After downloading all the files needed for your version of the game, navigate to \HigurashiEp0X_Data\StreamingAssets (where X is the chapter number)
    Create a folder called voice
    Open the voices zip, then the SE folder inside it and extract the folders inside the voice folder you created in the previous step.

    (optional) Extract the MangaGamer sprites patch CGAlt folder inside the StreamingAssets folder. Merge and overwrite the files if asked.

    Extract the graphics patch CG folder inside the StreamingAssets folder. Merge and overwrite the files if asked.
    Extract the textboxes patch CG folder inside the StreamingAssets folder. Merge and overwrite the files if asked.
    Extract the patch Update, SE and CG folders (when available) inside StreamingAssets folder. Merge and overwrite the files if asked.
    Navigate back to \HigurashiEp0X_Data (where X is the chapter number) and extract the patch StreamingAssets and Managed folders inside this folder.


>

**Comment by P-Chang** [comment]
@RetroGameClub 

http://www.mediafire.com/file/2i4alggutcf3vh1/Chapter2DLL_FOR_MGVer.rar
probably it works well.

by the way, your onikakushi with patched dll really works with only StreamingAssets folder overwriting?
I can't believe that works with MG version.

**Comment by RetroGameClub** [comment]
@P-Chang that dll from mediafire works, the game now plays but the voices aren't playing; the zipped files I downloaded for the patch are the Textboxes, the Voice + Graphics patch V3, the CG, the CGAlt, and the Voices

**Comment by P-Chang** [comment]
@RetroGameClub 
Can you upload screenshot that show the your voice folder in explore and property of voice folder?

Or sometimes voice volume set to 0 automatically, press LeftShift + M once time.

**Comment by RetroGameClub** [comment]
@P-Chang LeftShift +M did nothing, but here are the screenshots
![watana2](https://user-images.githubusercontent.com/32254569/32138269-9415b882-bbe4-11e7-85a3-991641d813a4.png)
![watana](https://user-images.githubusercontent.com/32254569/32138272-971e7366-bbe4-11e7-967e-e3c1d64b3113.png)
![watana1](https://user-images.githubusercontent.com/32254569/32138273-9a3b1572-bbe4-11e7-9c21-63901974dbf4.png)

**Comment by DoctorDiablo** [comment]
Shouldn't the voice folder be inside the StreamingAssets folder?

**Comment by EliEron** [comment]
@RetroGameClub The "voice" folder needs be placed inside the "StreamingAssets" folder, not outside it like it is in your screenshots.

If you move it inside "StreamingAssets" then the voices will likely work as the structure of the folder itself did seem to be correct.

Oops seems like @DoctorDiablo beat me to it. But yes to confirm it does indeed need to be the inside the "StreamingAssets" folder.

**Comment by RetroGameClub** [comment]
@EliEron its working now lol I knew voices belonged in the streamingassets folder I just totally hadnt noticed it wasn't there this time around as I've re-installed the game quite a few times, thanks for the help the new dll fixed the issue

**Comment by ItaloKnox** [comment]
@P-Chang the file you sent for Chapter 4 seems to be working fine. I tested it on Steam and drm-free, both working great. Steam is also supporting achievements and everything officially supported by MangaGamer on Steam, so we can say your changes were right.

A funny issue is that if you run the drm-free game using this dll while the Steam game is also installed, it will open both games at the same time. Before it used to open and close the drm-free game and then launch the Steam game, so your bypass seems to be working. We should leave this as it is, not really worth fixing this.

I'll let @enumag pull the trigger on these new dlls, if he gives the greenlight we can push them officially.

**Comment by enumag** [comment]
You have a green! :-)

**Comment by P-Chang** [comment]
@ItaloKnox 
http://www.mediafire.com/file/mwycypm6srly8yd/Chapter4_MG_Test2.rar

double launch issue is fixed easily. could you check it again?
it will happen in launch with exe on steam version.

**Comment by ItaloKnox** [comment]
@P-Chang 
You really fixed it, looks perfect now. It's still unlocking achievements and everything, so I think it looks good. Can you prepare the dlls for all chapters?

**Comment by P-Chang** [comment]
@ItaloKnox
I still have anxiety, but all DLL is here.
http://www.mediafire.com/file/f78fsrcd8dxg0vo/HigurashiDLL_MGfix.rar
